internal/controller/pipelinerollout_controller.go

```diff
@@ -559,23 +555,24 @@ func pipelineObservedGenerationCurrent(generation int64, observedGeneration int6
 
 // Set the Condition in the Status for child resource health
 
-func (r *PipelineRolloutReconciler) processPipelineStatus(ctx context.Context, pipelineRollout *apiv1.PipelineRollout) error {
+func (r *PipelineRolloutReconciler) processPipelineStatus(ctx context.Context, pipelineRollout *apiv1.PipelineRollout, existingPipelineDef *kubernetes.GenericObject) error {
 	numaLogger := logger.FromContext(ctx)
 
-	pipelineDef, err := r.makeRunningPipelineDefinition(ctx, pipelineRollout)
-	if err != nil {
-		return err
-	}
-
-	// Get existing Pipeline
-	// TODO: Eliminate the need for this call to GetLiveResource, instead use the existingPipelineDef from the reconcile
-	existingPipelineDef, err := kubernetes.GetLiveResource(ctx, r.restConfig, pipelineDef, "pipelines")
-	if err != nil {
-		if apierrors.IsNotFound(err) {
-			numaLogger.WithValues("pipelineDefinition", *pipelineDef).Warn("Pipeline not found. Unable to process status during this reconciliation.")
-		} else {
-			return fmt.Errorf("error getting Pipeline for status processing: %v", err)
+	// Only fetch the latest pipeline object while deleting the pipeline object, i.e. when pipelineRollout.DeletionTimestamp.IsZero() is false
+	if existingPipelineDef == nil {
+		pipelineDef, err := r.makeRunningPipelineDefinition(ctx, pipelineRollout)
+		if err != nil {
+			return err
+		}
+		livePipelineDef, err := kubernetes.GetLiveResource(ctx, pipelineDef, "pipelines")
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				numaLogger.WithValues("pipelineDefinition", *pipelineDef).Warn("Pipeline not found. Unable to process status during this reconciliation.")
```

Comment:
should we return if this happens? otherwise, `livePipelineDef` would be nil, right?